docker-app/qfieldcloud/subscription/models.py

```diff
@@ -819,70 +845,43 @@ def create_subscription(
-            ), "Creating a trial plan requires `active_since` to be a valid datetime object"
-
-            active_until = active_since + timedelta(days=config.TRIAL_PERIOD_DAYS)
-            logger.info(
-                f"Creating trial subscription from {active_since=} to {active_until=}"
-            )
-            trial_subscription = cls.objects.create(
-                plan=plan,
-                account=account,
-                created_by=created_by,
-                status=plan.initial_subscription_status,
-                active_since=active_since,
-                active_until=active_until,
+        current_time = timezone.now()
+
+        subscription_data = {
+            "account": account,
+            "created_by": created_by,
+            "status": SubscriptionStatus.ACTIVE_PAID,
+            "active_since": current_time,
+            "active_until": None,  # Stripe will set this
+            "regular_plan": regular_plan,
+            "trial_plan": None,
+            "trial_ends_at": None,
+        }
+
+        if trial_plan:
+            trial_end_date = current_time + timedelta(days=config.TRIAL_PERIOD_DAYS)
+
+            subscription_data.update(
+                {
+                    "trial_plan": trial_plan,
+                    "trial_ends_at": trial_end_date,
+                }
             )
-            # NOTE to get annotations, mostly `is_active`
-            trial_subscription_obj = cls.objects.get(pk=trial_subscription.pk)
 
             if (
                 account.user.is_organization
                 and account.user.organization_owner.remaining_trial_organizations > 0
             ):
-                account.user.organization_owner.remaining_trial_organizations -= 1
-                account.user.organization_owner.save(
-                    update_fields=["remaining_trial_organizations"]
-                )
-
-            # the trial plan should be the default plan
-            regular_plan = Plan.objects.get(
-                user_type=account.user.type,
-                is_default=True,
-            )
-
-            # the end date of the trial is the start date of the regular
-            regular_active_since = active_until
-        else:
-            trial_subscription_obj = None
-            regular_plan = plan
-            regular_active_since = active_since
-
-            # NOTE in case the user had a custom amount set (e.g manually set by support) this will
-            # be overwritten by a subscription plan change.
-            # But taking care of this would add quite some complexity.
-            if account.user.is_person:
-                account.user.remaining_trial_organizations = (
-                    regular_plan.max_trial_organizations
-                )
+                account.user.remaining_trial_organizations -= 1
                 account.user.save(update_fields=["remaining_trial_organizations"])
+            else:
+                raise NoRemainingTrialOrganizationsError
```

Comment:
```suggestion
                raise NoRemainingTrialOrganizationsError()
```

In the codebase we never throw a class, we always instantiate error classes. Valid also for JS.

Reply:
Indeed, thanks!